server/game/gameSystems/DiscardFromDeckSystem.ts

```diff
@@ -0,0 +1,82 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName } from '../core/Constants';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import Player from '../core/Player';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+import * as Contract from '../core/utils/Contract';
+
+export interface IDiscardFromDeckProperties extends IPlayerTargetSystemProperties {
+    amount?: number;
+}
+
+export class DiscardFromDeckSystem<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardFromDeckProperties> {
+    public override readonly name = 'discardFromDeck';
+    public override readonly eventName = EventName.OnDiscardFromDeck;
+
+    protected override defaultProperties: IDiscardFromDeckProperties = {
+        amount: 1
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public override eventHandler(_event): void { }
+
+    // public eventHandler(event): void {
+    //     for (let i = 0; i < event.amount; i++) {
+    //         const topCard = event.player.getTopCardOfDeck();
+    //         event.player.moveCard(topCard, Location.Discard);
+    //     }
+    // }
+
+    public override getEffectMessage(context: TContext): [string, any[]] {
+        const properties = this.generatePropertiesFromContext(context);
+        return ['discard ' + (properties.amount + (properties.amount > 1 ? ' cards' : ' card') + ' from deck'), []];
+    }
+
+    public override canAffect(player: Player, context: TContext, additionalProperties = {}): boolean {
+        const properties = this.generatePropertiesFromContext(context, additionalProperties);
+        return properties.amount !== 0 && super.canAffect(player, context);
```

Comment:
Oh yeah there's a lot of nuance here about whether this should report back true or false if there aren't enough cards in the deck, because in some situations it's okay if an effect resolves and in some cases it isn't. Feel free to ask in chat if you want details but the tl;dr is you can just reuse the logic from DiscardCardsFromHand in terms of mustChangeGameState and isCost.